addons/web/tests/test_session_info.py

```diff
@@ -14,7 +14,9 @@ def setUpClass(cls):
         cls.company_a = cls.env['res.company'].create({'name': "A"})
         cls.company_b = cls.env['res.company'].create({'name': "B"})
         cls.company_c = cls.env['res.company'].create({'name': "C"})
-        cls.companies = [cls.company_a, cls.company_b, cls.company_c]
+        cls.company_b_branch = cls.env['res.company'].create({'name': "B Branch", 'parent_id': cls.company_b.id})
+        cls.allowed_companies = [cls.company_a, cls.company_b_branch, cls.company_c]
+        cls.disallowed_ancestor_companies = [cls.company_b]
```

Comment:
Please use recordsets, as this enables the whole recordset API
```suggestion
        cls.allowed_companies = cls.company_a + cls.company_b_branch + cls.company_c
        cls.disallowed_ancestor_companies = cls.company_b
```